app/Database/CourseQueries.hs

```diff
@@ -269,3 +274,53 @@ queryGraphs =
     runSqlite databasePath $
         do graphs :: [Entity Graph] <- selectList [] [Asc GraphTitle]
            return $ createJSONResponse graphs
+
+-- ========================================================
+
+-- getLectureTime :: (String, String, String) -> IO ([Time])
+-- getLectureTime (code, section, session) = do
+--     maybeEntityLectures  <- selectFirst [LectureCode ==. (T.pack code),
+--                                          LectureSection ==. (T.pack section),
+--                                          LectureSession ==. (T.pack session)]
+--                                         []
+--     case maybeEntityLectures of
+--         Nothing -> return []
+--         Just entityLectures -> return $ lectureTimes . entityVal $ entityLectures
+
+
+-- getLectureTime :: CourseInfo -> IO (CourseInfo)
+-- getLectureTime :: (MonadIO m, Control.Monad.Reader.Class.MonadReader env m,
+--                          HasPersistBackend env SqlBackend) =>
+--                         CourseInfo -> m CourseInfo
+getLectureTime courseInfo = do
+    maybeEntityLectures <- selectFirst [LectureCode ==. (T.pack . code $ courseInfo),
+                                        LectureSection ==. (T.pack . section $ courseInfo),
+                                        LectureSession ==. (T.pack . session $ courseInfo)]
+                                       []
+    -- liftIO $ print maybeEntityLectures
+    case maybeEntityLectures of
+        Nothing -> return CourseInfo {code = code courseInfo, section = section courseInfo, session = session courseInfo, time = []}
```

Comment:
Use a `let time = ...` to first extract the time, and then you can remove the code duplication in the `return`